src/PSR7/ServerRequestValidator.php

```diff
@@ -49,6 +49,7 @@ public function getSchema() : OpenApi
     /**
      * @return OperationAddress which matched the Request
      *
+     * @throws MultipleOperationsMismatchForRequest
```

Comment:
There is no need in it

https://github.com/thephpleague/openapi-psr7-validator/blob/4b45a7aac780429571cbb8c4ccfef6983ab560be/src/PSR7/Exception/MultipleOperationsMismatchForRequest.php#L12

which is already declared

Reply:
Interesting, will have a check tomorrow why PHPStorm didn't pick this up corretcly :+1:

Reply:
Sorry for the delay. So I have this bit of (WIP) code where both the exceptions are caught and handled differently:

```php
        try {
            $operationAddress = $this->requestValidator->validate($request);
            return resolve($next($request))->then(function (ResponseInterface $response) use ($operationAddress) {
                try {
                    $this->responseValidator->validate($operationAddress, $response);
                } catch (ValidationFailed $validationFailed) {
                    //
                }
            });
        } catch (MultipleOperationsMismatchForRequest $multipleOperationsMismatchForRequest) {
            return $next($request);
        } catch (ValidationFailed $validationFailed) {
            //
        }
```

Without adding `@throws MultipleOperationsMismatchForRequest` PHPStorm complains that the exception is never thrown in the called method:

![image](https://user-images.githubusercontent.com/147145/66204032-29b3f880-e6aa-11e9-84e7-6cdffbbe1459.png)

Reply:
I hope this could be considered a bug in inspections

Reply:
Probably, will do some more testing and raise an issue with JetBrains about this.